[export] Log module hierarchy.

zhxchen17 opened the pull request:
Summary:
We can also log the module hierarchy in the following format:
```
:ToplevelModule
sparse:SparshArch
dense:DenseArch
```
So that we can have more information recorded about model's identity.

Test Plan: CI

Differential Revision: D54921097

Skylion007 commented on torch/export/_trace.py:
```diff
@@ -312,6 +313,30 @@ def _restore_state_dict(
 
 
+def _get_module_hierarchy(mod: torch.nn.Module) -> Dict[str, str]:
+    ret = {}
```
nit can be a dict comprehension

zhxchen17 committed:
[export] Log module hierarchy. (#121970)

Summary:

We can also log the module hierarchy in the following format:
```
:ToplevelModule
sparse:SparshArch
dense:DenseArch
```
So that we can have more information recorded about model's identity.

Test Plan: CI

Differential Revision: D54921097

izaitsevfb commented:
@angelayi , could you please add relevant label and re-merge?

zhxchen17 commented:
@pytorchbot merge

zhxchen17 commented:
@pytorchbot merge

huydhn commented:
@pytorchbot merge -i
